modules/nf-google/src/main/nextflow/cloud/google/lifesciences/GoogleLifeSciencesHelper.groovy

```diff
@@ -186,8 +186,15 @@ class GoogleLifeSciencesHelper {
                 .setServiceAccount(serviceAccount)
                 .setPreemptible(req.preemptible)
 
+        def network = new Network()
         if( req.usePrivateAddress ) {
-            vm.setNetwork( new Network().setUsePrivateAddress(true) )
+            network.setUsePrivateAddress(true)
+        }
+        if( req.network ) {
+            network.setNetwork(req.network)
+        }
+        if(req.network!=null | req.usePrivateAddress ) {
```

Comment:
I think the condition should `req.network || req.usePrivateAddress` (not null it's evaluated to true by groovy and `|` it's bitwise OR)